src/main/java/org/prebid/server/privacy/gdpr/tcfstrategies/purpose/typestrategies/FullEnforcePurposeStrategy.java

```diff
@@ -84,11 +84,12 @@ private RestrictionType restrictionType(VendorPermissionWithGvl vendorPermission
         final VendorPermission vendorPermission = vendorPermissionWithGvl.getVendorPermission();
         final Integer vendorId = vendorPermission.getVendorId();
 
-        return publisherRestrictions.stream()
+        // the vendorId is null if the bidder is not enabled
+        return vendorId != null ? publisherRestrictions.stream()
                 .filter(publisherRestriction -> publisherRestriction.getVendorIds().contains(vendorId))
                 .map(PublisherRestriction::getRestrictionType)
                 .findFirst()
-                .orElse(RestrictionType.UNDEFINED);
+                .orElse(RestrictionType.UNDEFINED) : RestrictionType.UNDEFINED;
```

Comment:
This will work. But lets replace this approach with 
```
 final List<VendorPermissionWithGvl> validVendorForPurpose = vendorsForPurpose.stream()
                .filter(vendorPermissionWithGvl -> vendorPermissionWithGvl.getVendorPermission().getVendorId() != null)
                .collect(Collectors.toList());
```

after 
```
final List<VendorPermission> allowedExcluded = allowedExcludedVendorPermission(excludedVendors,
                publisherRestrictions);
```
(and use this value in `mapVendorPermission`)

To be more explicit about that.

Reply:
Will do :smile_cat: